lib/user_comm_screener.rb

```diff
@@ -193,6 +237,21 @@ def load_preference_map
     UserCommPrefs.new(acting_user, resolved_user_communication_preferences)
   end
 
+  def actor_preferences
+    @actor_preferences ||= begin
+      user_ids_by_preference_type = actor_communication_preferences.reduce({}) do |hash, pref|
+        hash[pref.preference_type] ||= []
+        hash[pref.preference_type] << pref.target_user_id
+        hash
+      end
```

Comment:
```suggestion
      user_ids_by_preference_type = actor_communication_preferences.reduce(Hash.new([])) do |hash, pref|
        hash[pref.preference_type] << pref.target_user_id
        hash
      end
```

I learnt a new trick from @OsamaSayegh recently.

Reply:
Ah yeah that is better. Really wish there was a simple ruby method for this, like `make_hash_from_array_pls`